### PR TITLE
OSD-15841: Check proxy and API connection if login failed

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
### What type of PR is this?

feature

### What this PR does / Why we need it?

Check the API connection via HTTP proxy transmission and provide a proper error message to the end user 

### Which Jira/Github issue(s) does this PR fix?

[OSD-15841](https://issues.redhat.com//browse/OSD-15841)

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [] Included documentation changes with PR

### CODE REVIEW

File: pkg/cli/config/config.go [hunk 1]
```diff
@@ -70,3 +74,46 @@ func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
...
 	return bpConfig, nil
 }
+
+// CheckAPIConnection validate API connection via configured proxy and VPN
+func (config BackplaneConfiguration) CheckAPIConnection() error {
+
+	// Check backplane Proxy URL
+	if config.ProxyURL == "" {
+		//Cannot connect to backplane API URL: Check if you need to use a proxy/VPN to access backplane
+		return errors.New("empty proxy url - check your backplane-cli configuration in $HOME/.config/backplane/config.json")
```
Comment (bmeng) [inline pkg/cli/config/config.go:84]: The backplane config file could be other if it is set by env?

File: pkg/cli/config/config.go [hunk 2]
```diff
@@ -70,3 +74,46 @@ func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
...
+
+	if !connectionOk {
+		return err
+	}
+
+	return nil
+}
+
+// pingBackplaneAPIviaProxy returns status of the the API connection
+func (config BackplaneConfiguration) pingBackplaneAPIviaProxy() (bool, error) {
```
Comment (bmeng) [inline pkg/cli/config/config.go:98]: Can we rename the function name here? As the implementation is to send a normal http request, the `ping` here might be a bit confusing.

File: pkg/cli/config/config.go (hunk 1 above)
Comment (samanthajayasinghe) [reply pkg/cli/config/config.go]: Thanks @bmeng , Derive the path dynamically

File: pkg/cli/config/config.go (hunk 2 above)
Comment (samanthajayasinghe) [reply pkg/cli/config/config.go]: Rename the function to `testHttpRequestToBackplaneAPI()`, but let me know if you have a better option  :)

**Pull request merged**